### ISSUE TITLE
No recommendation available

### ISSUE THREAD

**Issue by phkrief** [crossminer/scava#415]
I look for an example which could produce some recommendations.
For example, when I selct the following code:

> 		in = new URL(url).openStream();
> 		Reader reader = new InputStreamReader(in);
> 		String line;
> 	        BufferedReader read = new BufferedReader(reader);

And "Request code recommendation" I receive the following message:
![image](https://user-images.githubusercontent.com/6196602/67939528-0d847800-fbd2-11e9-8efc-e9ba26174e97.png)
What kind of example should I take to have a chance to get some recommendations?

Thanks a lot

**Comment by phkrief** [comment]
@md2manoppello Now that we can reach the KB, I'm still struggling with this feature.  Actually, I need to have written almost the whole example to get some snipped code. 
What is the idea behind these snipped codes? How does it work?
Thanks for your help

**Comment by davidediruscio** [comment]
Hi @phkrief, isn't the step-by-step user guide that @md2manoppello provided you with detailed enough?

**Comment by phkrief** [comment]
Hi @davidediruscio ,
Juri document is great. Now, I'm going through it to understand how everything works and, in some cases, I have a concern or a question and I need some explanation.
For example, for this recommendation, when I reach the step mentioned by Juri, I already have all my code written so I don't understand why I would need more recommendations unless Juri wants to emphasis something specific.
This is my concern about that feature.

Now, I could make it work, I also tried another scenario: what would be the minimum of information I need to provide get such useful snipped code. In my case, I need to provide almost all the code. So, it might mean that this is not a good example.

In other words, yes, I use and appreciate Juri document but I also try to understand :-).

Does it make more sense now ?

**Comment by davidediruscio** [comment]
@phkrief can we close this issue?

**Comment by phkrief** [comment]
Yes, we can! I'm very happy to do it ;-)
